sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1673,6 +1673,120 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-39557 INSERT INTO statements with tables with array/struct/map defaults") {
```

Comment:
nit: let split the test case into 3 test cases

Reply:
Sounds good, done.